community/values/src/main/java/org/neo4j/values/storable/PointValue.java

```diff
@@ -280,9 +280,32 @@ public static PointValue fromMap( MapValue map )
     private static Pattern quotesPattern = Pattern.compile( "^[\"']|[\"']$" );
 
     public static PointValue parse( CharSequence text )
+    {
+        return PointValue.parse( text, null );
+    }
+
+    public static PointValue parse( CharSequence text, AnyValue[] fieldsFromHeader )
+    {
+        AnyValue[] fieldsFromData = parseIntoArray( text );
+        if ( fieldsFromHeader != null )
+        {
+            //It is given that fieldsFromData.length == fieldsFromHeader.length because parseIntoArray produces fixed length arrays
+            // Merge InputFields: Data fields override header fields
+            for ( int i = 0; i < fieldsFromData.length; i++ )
+            {
+                if ( fieldsFromData[i] == null )
+                {
+                    fieldsFromData[i] = fieldsFromHeader[i];
+                }
+            }
+        }
+        return fromInputFields( fieldsFromData );
+    }
+
+    public static AnyValue[] parseIntoArray( CharSequence text )
```

Comment:
We could encapsulate this as a `PointValueFields` (static inner) class if we make this public. Then we also know we have the correct length when we pass it in as a `defaultFields` argument.